packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/advanced_marker.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show Offset;
+import 'package:flutter/foundation.dart';
+
+import '../../google_maps_flutter_platform_interface.dart';
+
+/// Marks a geographical location on the map.
+///
+/// Extends [Marker] and provides additional features
+@immutable
+class AdvancedMarker extends Marker {
+  /// Creates a set of marker configuration options.
+  ///
+  /// Default marker options.
+  ///
+  /// Specifies a marker that
+  /// * is fully opaque; [alpha] is 1.0
+  /// * uses icon bottom center to indicate map position; [anchor] is (0.5, 1.0)
+  /// * has default tap handling; [consumeTapEvents] is false
+  /// * is stationary; [draggable] is false
+  /// * is drawn against the screen, not the map; [flat] is false
+  /// * has a default icon; [icon] is `BitmapDescriptor.defaultMarker`
+  /// * anchors the info window at top center; [infoWindowAnchor] is (0.5, 0.0)
+  /// * has no info window text; [infoWindowText] is `InfoWindowText.noText`
+  /// * is positioned at 0, 0; [position] is `LatLng(0.0, 0.0)`
+  /// * has an axis-aligned icon; [rotation] is 0.0
+  /// * is visible; [visible] is true
+  /// * is placed at the base of the drawing order; [zIndex] is 0.0
+  /// * reports [onTap] events
+  /// * reports [onDragEnd] events
+  /// * is always displayed on the map regardless of collision with other
+  /// markers; [collisionBehavior] is [MarkerCollisionBehavior.required]
+  const AdvancedMarker({
+    required super.markerId,
+    super.alpha,
+    super.anchor,
+    super.consumeTapEvents,
+    super.draggable,
+    super.flat,
+    super.icon,
+    super.infoWindow,
+    super.position,
+    super.rotation,
+    super.visible,
+    super.zIndex,
+    super.clusterManagerId,
+    super.onTap,
+    super.onDrag,
+    super.onDragStart,
+    super.onDragEnd,
+    this.collisionBehavior = MarkerCollisionBehavior.required,
+  });
```

Comment:
I don't think it's needed here. Since `AdvancedMarker` is a `Marker`, it still works with other bitmap descriptors. I'd like to add an assertion about `Marker` not using `PinConfig` but didn't find the right place for it